developers/weaviate/concepts/indexing/hnsw-indexes.mdx

```diff
@@ -0,0 +1,156 @@
+---
+title: HNSW Indexes
+sidebar_position: 20
+image: og/docs/indexing.jpg
+# tags: ['basics']['indexes']
+---
+
+import HNSWIntro from '/_includes/indexes/hnsw-intro.mdx';
+
+<HNSWIntro/>
+
+import HNSWHow from '/_includes/indexes/hnsw-how.mdx';
+
+<HNSWHow/>
+
+To configure an HNSW index, see: [Configure HNSW indexes](/developers/weaviate/configuration/indexing-vector/hnsw-indexes).
+
+## Overview
+
+HNSW is an algorithm that works on multi-layered graphs. It is also an index. An HNSW index is the vector index that Weaviate creates when it applies the HNSW algorithm to the vectors in a collection. Weaviate's [implementation](https://github.com/weaviate/weaviate/tree/main/adapters/repos/db/vector/hnsw) offers full [CRUD-support](https://db-engines.com/en/blog_post/87).
+
+When Weaviate builds the index, the HNSW algorithm creates a series of layers. The layers improve search because the database engine doesn't have to scan the entire vector space at query time. Instead, the HNSW algorithm uses the layers to build a list of approximate nearest neighbors (ANN) quickly and efficiently.
+
+### Example - HNSW search
+
+Consider this diagram of a vector search that uses HNSW.
+
+![HNSW layers](../img/hnsw-layers.svg "HNSW layers")
+
+Layer zero is the lowest layer. Layer zero contains every object in the database, and the objects are well connected to each other.
+
+Some of the objects are also represented in the layers above layer zero. Each layer above layer zero has fewer objects and fewer connections.
+
+When HNSW searches the graph, it starts at the highest layer. The algorithm finds the closest matching data points in the highest layer. Then, HNSW goes one layer deeper, and finds the closest matching data points in the lower layer that correspond to the objects in the higher layer. These are the nearest neighbors.
```

Comment:
> Then, HNSW goes one layer deeper, and finds the closest matching data points in the lower layer that correspond to the objects in the higher layer. These are the nearest neighbors.

I believe it traverses downwards in each graph, and builds a "candidate list" - determined by `ef` or `dynamicEf` factors.